include/mesh/mesh_base.h

```diff
@@ -852,6 +854,8 @@ class MeshBase : public ParallelObject
   virtual const_node_iterator local_nodes_end    () const = 0;
   virtual const_node_iterator pid_nodes_begin    (const processor_id_type proc_id) const = 0;
   virtual const_node_iterator pid_nodes_end      (const processor_id_type proc_id) const = 0;
+  virtual const_node_iterator bid_nodes_begin    (const boundary_id_type bndry_id) const = 0;
```

Comment:
This is probably a good place to break our long-standing "don't properly document iterator range methods" tradition.  I'd have assumed that the behavior here was "iterate through nodes that are on sides with the given boundary id", but that only happens if the nodes themselves also share that id, right?  We should probably clarify that with a comment.